File: 프로그래머스/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.js
function solution(lottos, win_nums) {
    
    const rank = [6, 6, 5, 4, 3, 2, 1];
    
    const set = new Set();
    
    for(let i=0; i<win_nums.length; i++){
        set.add(win_nums[i]);
    }
    
    var cnt = 0;
    var zero = 0;
    
    for(let i=0; i<lottos.length; i++){
        if(set.has(lottos[i])) cnt++;
        if(lottos[i] == 0) zero++;
    }
    
    var answer = [rank[cnt+zero], rank[cnt]];
    
    return answer;
}

File: 프로그래머스/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.java
import java.util.*;

class Solution {
    public int[] solution(int[] lottos, int[] win_nums) {
        
        int[] rank = {6, 6, 5, 4, 3, 2, 1};
        
        HashSet<Integer> set = new HashSet<>();
        
        for(int i=0; i<win_nums.length; i++){
            set.add(win_nums[i]);
        }
        
        int cnt = 0;
        int zero = 0;
        
        for(int i=0; i<lottos.length; i++){
            if(set.contains(lottos[i])) cnt++;
            if(lottos[i] == 0) zero++;
        }
        
        int[] answer = {rank[cnt+zero], rank[cnt]};
        
        return answer;
    }
}